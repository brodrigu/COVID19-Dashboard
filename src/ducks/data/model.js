import { createAction, handleActions } from 'redux-actions';
import { put, takeEvery } from 'redux-saga/effects';
import { calculateData } from '~services/covid-model';

// Action creators
export const calculateModelData = createAction('data/model/calculateModelData');
export const updateModelData = createAction('data/model/updateModelData');

// Reducers
export default handleActions({
    [updateModelData](state, { payload }) {
        return payload;
    },
}, []);

// Sagas

// list
export function* workCalculateModelData() {
    yield put(updateModelData(calculateData({ daysToCalculate: 100 })));
}
export function* watchCalculateModelData() {
    yield takeEvery([calculateModelData], workCalculateModelData);
}
