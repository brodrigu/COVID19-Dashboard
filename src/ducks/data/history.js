import { createAction, handleActions } from 'redux-actions';
import { put, takeEvery } from 'redux-saga/effects';
import { getDailyUSNumbers } from '~clients/covid-tracking';

// Action creators
export const requestHistoricalData = createAction('data/history/requestHistoricalData');
export const updateMHistoricalData = createAction('data/history/HistoricalData');

// Reducers
export default handleActions({
    [updateMHistoricalData](state, { payload }) {
        return payload;
    },
}, []);

// Sagas

// list
export function* workRequestHistoricalData() {
    const data = yield getDailyUSNumbers();
    yield put(updateMHistoricalData(data));
}
export function* watchRequestHistoricalData() {
    yield takeEvery([requestHistoricalData], workRequestHistoricalData);
}
