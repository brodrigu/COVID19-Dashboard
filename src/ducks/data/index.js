import { combineReducers } from 'redux';
import { all } from 'redux-saga/effects';

import model, { watchCalculateModelData } from './model';

export default combineReducers({
    model,
});

export function* dataSagas() {
    yield all([
        watchCalculateModelData(),
    ]);
}
