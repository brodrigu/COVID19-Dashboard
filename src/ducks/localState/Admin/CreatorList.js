import { createAction, handleActions } from 'redux-actions';
import { takeEvery } from 'redux-saga/effects';
import { fetchCreatorListRequest } from '../../data/creatorList';

// Action creators
export const setPageNum = createAction('Admin/CreatorList/setPageNum');

const initialState = {
    pageNum: 1,
};

// Reducers
export default handleActions({
    [setPageNum](state, { payload }) {
        return { ...state, pageNum: payload };
    },
}, initialState);

export function* watchFetchCreatorListRequest() {
    yield takeEvery([setPageNum], fetchCreatorListRequest);
}
