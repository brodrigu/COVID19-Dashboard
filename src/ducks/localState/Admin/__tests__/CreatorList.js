/* eslint-env jest */
import configureStore from 'redux-mock-store';
import { takeEvery } from 'redux-saga/effects';

// Actions to be tested
import * as CreatorListDuck from '../CreatorList';
import { fetchCreatorListRequest } from '../../../data/creatorList';

const mockStore = configureStore();
const store = mockStore();

describe('CreatorList Actions', () => {
    beforeEach(() => {
        store.clearActions();
    });

    for (const action of [
        'setPageNum',
    ]) {
        test(`${action} test`, () => {
            store.dispatch(CreatorListDuck[action]());
            expect(store.getActions()).toMatchSnapshot();
        });
    }
});

describe('CreatorList Reducers', () => {
    beforeEach(() => {
    });

    test('initial state', () => {
        const action = { type: 'dummy_action' };
        expect(CreatorListDuck.default(undefined, action)).toMatchSnapshot();
    });

    test('setPageNum', () => {
        const action = { payload: 4, type: `${CreatorListDuck.setPageNum}` };
        expect(CreatorListDuck.default(undefined, action)).toMatchSnapshot();
    });
});


describe('CreatorList Sagas', () => {
    beforeEach(() => {
    });

    test('watchFetchCreatorListRequest', () => {
        const gen = CreatorListDuck.watchFetchCreatorListRequest();
        expect(gen.next().value).toEqual(takeEvery(
            [CreatorListDuck.setPageNum],
            fetchCreatorListRequest,
        ));
        expect(gen.next().done).toEqual(true);
    });
});
