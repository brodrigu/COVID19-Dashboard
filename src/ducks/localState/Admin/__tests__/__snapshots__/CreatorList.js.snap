// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatorList Actions setPageNum test 1`] = `
Array [
  Object {
    "type": "Admin/CreatorList/setPageNum",
  },
]
`;

exports[`CreatorList Reducers initial state 1`] = `
Object {
  "pageNum": 1,
}
`;

exports[`CreatorList Reducers setPageNum 1`] = `
Object {
  "pageNum": 4,
}
`;
